src/main/database.cpp

```diff
@@ -68,6 +84,21 @@ void Database::setLoggingLevel(std::string loggingLevel) {
     spdlog::set_level(LoggingLevelUtils::convertStrToLevelEnum(std::move(loggingLevel)));
 }
 
+void Database::checkAccessMode() {
+    if (!FileUtils::fileOrPathExists(databasePath)) {
```

Comment:
I think we should check if the access mode is set to READ_ONLY, if so, we should throw an exception "Cannot create an empty database under READ ONLY mode."